### ISSUE TITLE
bug: v7 ion-loading isOpen property sometimes stuck and ignored on fast change

### ISSUE THREAD

**Issue by dobripet** [ionic-team/ionic-framework#27200]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

Hello, 
When using Angular ion-loading component I noticed that when there is fast change in isOpen value sometimes the loader gets stuck in wrong state.
See example screen 

![image](https://user-images.githubusercontent.com/11678095/231816462-dea28bec-1467-452e-b59f-9ff898706545.png)

Thank you for looking into that.

### Expected Behavior

Loader is shown when isOpen is true.
Loader is hidden when isOpen is false.

### Steps to Reproduce

Forked docs example 

https://stackblitz.com/edit/angular-lwgtrv?file=src/app/example.component.ts

click few times and it should happen.

### Ionic Info

Forked docs example on Stackblitz, not sure how to do this there...

**Comment by mapsandapps** [comment]
Thank you for the bug report and reproduction!

**Comment by gianbasagre** [comment]
This is an issue for us as well. It would be good to get this fixed as the inline method is the recommended way from the official docs.

**Comment by jvanderberg** [comment]
We are getting this with Ionic React as well. It periodically causes our tests to fail because the popup is modal and blocks access to the page behind, which has actually loaded. It's very intermittent.

**Comment by ryan-ju** [comment]
We're facing the same issue when using `IonLoading`, where parent unmount doesn't clear the loading modal.

Looking at the [source code](https://github.com/ionic-team/ionic-framework/blob/721fc6fd9a21c9c03bb21d290e9b1bbd703e43e3/packages/react/src/components/createInlineOverlayComponent.tsx#L69), `IonLoading` internally uses a class component.  I guess the bug is around the bridging code between functional and class components.

**Comment by Ferdysd96** [comment]
I had the same problem with the `isOpen` attribute, the solution I did to solve this problem is the following I hope it helps someone !!

```
import { Injectable } from '@angular/core';
import { LoadingController, LoadingOptions } from '@ionic/angular';


@Injectable({
    providedIn: 'root',
})
export class LoadingService {
    public isOPen: boolean = false;
    private loading!: HTMLIonLoadingElement;
    private initialOPts: LoadingOptions = { message: 'Loading...' };

    constructor(private loadingCtrl: LoadingController) {
        this.init(this.initialOPts)
    }

    async init(options: LoadingOptions) {
        this.loading = await this.loadingCtrl.create(options);
    }

    present(): void {
        this.isOPen = true;
        this.loading.present();
    }

    dismiss(): void {
        if (this.isOPen) {
            this.loading.dismiss();
            this.isOPen = false;
        }
    }

}
```

**Comment by dobripet** [comment]
I can confirm that this can be workaround. Did some of that last week but forgot to post it here.
Custom didPresent check that dismiss if should be closed and ngdestroy implementation that dismiss it worked for us so far...

**Comment by mapsandapps** [comment]
Hi, thank you for submitting this bug. It has been resolved via #28069, and the fix will be released in an upcoming version of Ionic Framework.
